### PR TITLE
Constanst.setRefValue() was always null

### PR DESCRIPTION
**Pull request by fjtirado**
Setting the ref value when constant is an uri to allow implementations
to try load them in case default loading fails

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]
@tsurdilo I think we can set up a formatting profile for the SDK (see the imports). I'll do it. So, with a `mvn clean install` the files are automatically formatted.

**Comment by fjtirado** [comment]
> @tsurdilo I think we can set up a formatting profile for the SDK (see the imports). I'll do it. So, with a `mvn clean install` the files are automatically formatted.

I remove the changed imports, but yes, template is a nice idea, thanks @ricardozanini

**Comment by tsurdilo** [comment]
Thanks @fjtirado , could we please add a test for this change as well?

**Comment by tsurdilo** [comment]
@manick02 can you please work with @fjtirado on this pr? Let me know when its good to merge. Would be nice to:
1. add a test for it
2. maybe we should have a null check for constantsFileDef
3. whatever else you see might be needed
Thanks!

**Comment by manick02** [comment]
> @manick02 can you please work with @fjtirado on this pr? Let me know when its good to merge. Would be nice to:
> 
> 1. add a test for it
> 2. maybe we should have a null check for constantsFileDef
> 3. whatever else you see might be needed
>    Thanks!

sure @tsurdilo

**Comment by manick02** [comment]
> @tsurdilo I think we can set up a formatting profile for the SDK (see the imports). I'll do it. So, with a `mvn clean install` the files are automatically formatted.

We have this plugin already in java sdk @ricardozanini 
           <plugin>
                    <groupId>com.coveo</groupId>
                    <artifactId>fmt-maven-plugin</artifactId>
                    <version>${version.fmt-maven-plugin}</version>
                </plugin>

**Comment by fjtirado** [comment]
@tsurdilo 
Test added

File: api/src/test/java/io/serverlessworkflow/api/test/MarkupToWorkflowTest.java [hunk 1]
```diff
@@ -569,6 +569,23 @@ public void testConstants(String workflowLocation) {
...
+    Workflow workflow = Workflow.fromSource(WorkflowTestUtils.readWorkflowFile(workflowLocation));
+
+    assertNotNull(workflow);
+    assertNotNull(workflow.getId());
+    assertNotNull(workflow.getName());
+    assertNotNull(workflow.getStates());
+
+    assertNotNull(workflow.getConstants());
+    Constants constants = workflow.getConstants();
+    assertNotNull(constants.getRefValue());
```
Comment (manick02) [inline api/src/test/java/io/serverlessworkflow/api/test/MarkupToWorkflowTest.java]: I think This assertion needs to be more strict to ensure the new behavior. recommend changing the assertion to assertEquals("  constantValues.json".constants.getRefValue());

File: api/src/main/java/io/serverlessworkflow/api/serializers/WorkflowSerializer.java [hunk 2]
```diff
@@ -159,8 +159,12 @@ public void serialize(Workflow workflow, JsonGenerator gen, SerializerProvider p
       gen.writeEndArray();
     }
 
-    if (workflow.getConstants() != null && !workflow.getConstants().getConstantsDef().isEmpty()) {
-      gen.writeObjectField("constants", workflow.getConstants().getConstantsDef());
+    if (workflow.getConstants() != null) {
```
Comment (cb-manick) [inline api/src/main/java/io/serverlessworkflow/api/serializers/WorkflowSerializer.java:162]: Is it possible to write test for this in WorkflowToMarkupTest?

Comment (fjtirado) [reply api/src/main/java/io/serverlessworkflow/api/serializers/WorkflowSerializer.java:162]: sure, done

File: api/src/test/java/io/serverlessworkflow/api/test/MarkupToWorkflowTest.java (hunk 1 above)
Comment (fjtirado) [reply api/src/test/java/io/serverlessworkflow/api/test/MarkupToWorkflowTest.java]: yes, better that way ;)

**Comment by manick02** [comment]
@tsurdilo Looks Good from my side

**Comment by tsurdilo** [comment]
@manick02 thanks!!

**Pull request merged**